src/script/common/c_converter.cpp

```diff
@@ -385,4 +385,13 @@ void setboolfield(lua_State *L, int table,
 	lua_setfield(L, table, fieldname);
 }
 
+void setstringfield(lua_State *L, int table,
+		const char *fieldname, const char *value)
+{
+	lua_pushstring(L, value);
+	if(table < 0)
+		table -= 1;
```

Comment:
I cleaner way to do this would be:
```
if (table < 0)
	table = lua_gettop(L) - table + 1;
lua_pushstring(L, value);
lua_setfield(L, table, field_name);
```
That's inconsistent with the other setters, but it doesn't break whenever you push or pop something.